packages/cdktf/test/variable.test.ts

```diff
@@ -168,3 +169,19 @@ test("nullable variable", () => {
   });
   expect(Testing.synth(stack)).toMatchSnapshot();
 });
+
+test("validation block variable", () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+
+  new TerraformVariable(stack, "test-variable", {
+    type: "string",
+    validation: [
+      {
+        errorMessage: "Validation failed..",
+        condition: Fn.regex("mickey_mouse", "mickey_mouse"),
```

Comment:
This doesn't seem like a very realistic test case. I'd suggest using one or both of the examples in the documentation: https://www.terraform.io/language/values/variables#custom-validation-rules

Reply:
The new test is much better, thank you.
I do still think we should actually reference the variable in the condition since that seems almost necessary for any practical usage of this. Unfortunately, I don't think a self reference during construction will work, so we may need an alternative way to specify it.

Reply:
@jsteinich Following the TF documentation, seems that it evaluates directly the variabile during the assertion phase.
During construction I think is not possible, as you said, to use the `this` keyword, but I don't have idea on how we could adress this. 
Maybe for the test is valuable something like `Fn.can(Fn.regex("^ami-", "${var.variable_name}"))` making possible the check during the `apply/deploy` command using directly the TF directives. 

(Sorry for ghosting here, I'm really sorry but I need to address a lot of stuff.. 😓 )

Reply:
No worries. I'll poke around a bit to try and come up with a pattern that makes sense.

Reply:
@bollohz sorry for taking so long to get back to this. I ended up just making it possible to add a validation after creation. That makes it possible to reference the created variable via `var.fqn`.

I thought about using something like `this` or `self` or `@var`, but it seemed too arbitrary and no clear sense of what it should be (especially when thinking about different languages).